feat: 마이페이지 계정 관리 닉네임 변경 기능 구현

gmlyb1 opened the pull request:
- 변경하기 버튼 클릭시 모달창을 통해 닉네임 변경값을 받아 변경

- 서버와 로컬스토리지 데이터 동시 변경 가능

-마이페이지 정보에 유저아이디와 이메일정보표시 -> 닉네임과 ,아이디, 이메일 정보표시로 변경

### 제목 : [feat: 마이페이지 계정 관리 닉네임 변경 기능 구현]
### PR 유형

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

### PR 체크리스트

<!-- PR이 다음 요구 사항을 충족하는지 확인하세요. -->

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

### 🔎 작업 내용

<!-- 어떤 작업을 하셨는지 상세하게 작성해주세요-->

- 변경하기 버튼 클릭시 모달창을 통해 닉네임 변경값을 받아 변경
- 서버와 로컬스토리지 데이터 동시 변경 가능
-마이페이지 정보에 유저아이디와 이메일정보표시 -> 닉네임과 ,아이디, 이메일 정보표시로 변경
  <br />

### 🔧 앞으로의 과제

- 내일 할 일을 적어주세요
  <br />

### 이슈

<!-- 이슈 키워드와 함께 #을 입력한 후 이슈 번호를 선택해주세요. -->
<!-- 에시 : resolves #1 -->

resolves #188

gmlyb1 committed:
마이페이지 계정 관리 닉네임 변경 기능 구현

- 변경하기 버튼 클릭시 모달창을 통해 닉네임 변경값을 받아 변경

- 서버와 로컬스토리지 데이터 동시 변경 가능

-마이페이지 정보에 유저아이디와 이메일정보표시 -> 닉네임과 ,아이디, 이메일 정보표시로 변경

devHyeon99 commented on src/pages/mypage/Mypage.jsx:
```diff
@@ -56,14 +66,20 @@ const Mypage = () => {
             </div>
             <nav aria-label="계정 관리">
-              <Link to="/mypage/setting" aria-label="계정 관리 페이지로 이동">
+              <button
```
Link 태그 수정

https://reactrouter.com/en/main/components/link#state

gmlyb1 replied:
해결

devHyeon99 commented on src/pages/mypage/Mypage.jsx:
```diff
@@ -1,31 +1,35 @@
+  const navigate = useNavigate(); // useNavigate 훅 사용
 
   useEffect(() => {
     const authData = localStorage.getItem('pocketbase_auth');
```
auth 키에 대한 값을 가져와서 데이터 렌더링 해주세요

gmlyb1 replied:
해결

devHyeon99 commented on src/pages/mypage/MyPageSetting.jsx:
```diff
@@ -1,81 +1,116 @@
   return (
     <section className="flex flex-col justify-between min-h-dvh pb-[80px]">
-      <TopHeader title="계정 관리" isShowIcon={true} />
+      <TopHeader title="계정 관리" isShowIcon />
```
true 값 빠짐

gmlyb1 replied:
해결

devHyeon99 commented on src/pages/mypage/MyPageSetting.jsx:
```diff
@@ -1,81 +1,116 @@
-  //     // API 호출을 통해 회원 탈퇴 처리
-  //     await pb.collection('users').delete(pb.authStore.model.id); // 현재 사용자 삭제
+    setIsLoading(true);
+    try {
```
로컬 스토리지에 있는 auth키의 user.name을 수정해줘야함

gmlyb1 replied:
해결

devHyeon99 commented on src/pages/mypage/MyPageSetting.jsx:
```diff
@@ -86,14 +121,45 @@ const MypageSetting = () => {
+      >
+        <div className="flex flex-col gap-4 p-4">
+          <h3 className="text-lg font-semibold">닉네임 변경</h3>
+          <input
```
라벨 필요

gmlyb1 replied:
해결

devHyeon99 commented on src/pages/mypage/MyPageSetting.jsx:
```diff
@@ -86,14 +121,45 @@ const MypageSetting = () => {
+          <button
+            onClick={checkNicknameAvailability}
+            className="bg-gray-300 text-white p-2 rounded-md"
+            aria-label="닉네임 중복 확인 버튼"
```
아리아 속성 제거 두개다!

gmlyb1 replied:
해결

devHyeon99 commented on src/pages/mypage/MyPageSetting.jsx:
```diff
@@ -1,81 +1,116 @@
           aria-labelledby="items-to-change-title"
         >
-          <div className="flex flex-col">
+          <div>
```
빠진 부분 다시 넣어주세요

gmlyb1 replied:
해결

gmlyb1 committed:
마이페이지 코드 리팩토링

- localstorage 값을 pocketbase__auth 에서 auth 값으로 변경해서 렌더링

- topHeader 컴포넌트에 isShowIcon true 빠져있던것 다시 원복

- 닉네임 변경시 데이터베이스 수정과 동시에 localstorage auth에도 해당값 세팅 처리

-input 근처 label 추가
-aria-label 필요없는것 삭제
-div className 빠진것 원복
